src/common/io/read.c

```diff
@@ -191,18 +192,18 @@ ioRead(IoRead *this, Buffer *buffer)
     // Store size of remaining portion of buffer to calculate total read at the end
     size_t outputRemains = bufRemains(buffer);
 
-    // Use any data in the output buffer left over from a line read
-    if (this->output != NULL && bufUsed(this->output) > 0 && bufRemains(buffer) > 0)
+    // Copy any data in the internal output buffer
+    if (this->output != NULL && bufUsed(this->output) - this->outputPos > 0 && bufRemains(buffer) > 0)
     {
-        // Determine how much data should be copied
-        size_t size = bufUsed(this->output) > bufRemains(buffer) ? bufRemains(buffer) : bufUsed(this->output);
+        // Internal output buffer remains taking into account the position
+        size_t outputRemains = bufUsed(this->output) - this->outputPos;
 
-        // Copy data to the user buffer
-        bufCatSub(buffer, this->output, 0, size);
+        // Determine how much data should be copied
+        size_t size = outputRemains > bufRemains(buffer) ? bufRemains(buffer) : outputRemains;
```

Comment:
can you rename outputRemains inside the IF to outputRemainsInternal or something? it would help clarify this section of code

Reply:
Done, and also done in ioReadLineParam().